### PR TITLE
PIN-4659 BKE - BFF - Add route reject purpose

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**
It needs [PIN-4658](https://github.com/pagopa/interop-be-purpose-process/pull/199) to compile

[PIN-4658]: https://pagopa.atlassian.net/browse/PIN-4658?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala [hunk 1]
```diff
@@ -133,6 +133,31 @@ final case class PurposesApiServiceImpl(
...
+
+    val result: Future[Unit] = for {
+      purposeUuid <- purposeId.toFutureUUID
+      versionUuid <- versionId.toFutureUUID
+      _           <- purposeProcessService.rejectPurposeVersion(
+        purposeUuid,
+        versionUuid,
+        PurposeProcess.RejectPurposeVersionPayload(payload.rejectionReason)
+      )
+      _           <- authorizationProcessService.removePurposeFromClients(purposeUuid)
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala]: This is not necessary.

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -10167,6 +10286,16 @@ components:
...
         type: string
   schemas:
+    RejectPurposeVersionPayload:
+      type: object
+      description: models the reject payload for this purpose version.
+      properties:
+        rejectionReason:
+          type: string
+          minimum: 10
+          maximum: 250
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: Constraints are in the process
```suggestion
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala [hunk 3]
```diff
@@ -133,6 +133,30 @@ final case class PurposesApiServiceImpl(
     pagination = Pagination(offset = offset, limit = limit, totalCount = pagedResults.totalCount)
   )
 
+  override def rejectPurposeVersion(purposeId: String, versionId: String, payload: RejectPurposeVersionPayload)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): akka.http.scaladsl.server.Route = {
+    logger.info(s"Rejecting purpose $purposeId with version $versionId")
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala]:
```suggestion
    logger.info(s"Rejecting version $versionId of purpose $purposeId")
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala [hunk 4]
```diff
@@ -133,6 +133,30 @@ final case class PurposesApiServiceImpl(
...
+      _           <- purposeProcessService.rejectPurposeVersion(
+        purposeUuid,
+        versionUuid,
+        PurposeProcess.RejectPurposeVersionPayload(payload.rejectionReason)
+      )
+    } yield ()
+
+    onComplete(result) {
+      val headers: List[HttpHeader] = headersFromContext()
+      handleError(s"Error rejecting purpose $purposeId with version $versionId", headers) orElse { case Success(_) =>
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala]:
```suggestion
      handleError(s"Error rejecting version $versionId of purpose $purposeId", headers) orElse { case Success(_) =>
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/PurposeProcessServiceImpl.scala [hunk 5]
```diff
@@ -48,6 +48,23 @@ class PurposeProcessServiceImpl(purposeProcessUrl: String, blockingEc: Execution
...
+      val request: ApiRequest[Unit] =
+        api.rejectPurposeVersion(
+          purposeId = purposeId,
+          versionId = versionId,
+          rejectPurposeVersionPayload = payload,
+          xCorrelationId = correlationId
+        )(BearerToken(bearerToken))
+      invoker.invoke(
+        request,
+        s"Reject Purpose $purposeId with version $versionId and reason ${payload.rejectionReason}"
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/PurposeProcessServiceImpl.scala]: Same, let's remove free text from logs
```suggestion
        s"Reject version $versionId of purpose $purposeId"
```

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**